From 10e1e19da099ded915c2e8a54d980a7851b5709f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Carlos=20Aznar=C3=A1n?= <caznaranl@uni.pe>
Date: Wed, 7 Feb 2024 20:31:00 -0500
Subject: [PATCH] Add support for OpenCL

---
 .SRCINFO           |  6 +++++-
 PKGBUILD           |  6 +++++-
 test_optdepends.sh | 11 +++++++++++
 3 files changed, 21 insertions(+), 2 deletions(-)

diff --git a/.SRCINFO b/.SRCINFO
index 5694525..095e259 100644
--- a/.SRCINFO
+++ b/.SRCINFO
@@ -37,11 +37,15 @@ pkgbase = petsc
 	optdepends = triangle: support for Triangle
 	optdepends = trilinos-ml: support for ML (part of Trilinos)
 	optdepends = zoltan: support for zoltan
+	optdepends = libyaml: support for YAML-formatted file
+	optdepends = opencl-headers: support for OpenCL
+	optdepends = ocl-icd: support for OpenCL
+	optdepends = sundials: for CVODE solver support
 	provides = petsc4py
 	options = staticlibs
 	source = http://web.cels.anl.gov/projects/petsc/download/release-snapshots/petsc-3.20.4.tar.gz
 	source = test_optdepends.sh
 	sha512sums = 68abd97377e1e30d3c09afa1617385e8ba56acc8508000c8da47112454462e9da35135076e678cea7a4fa6e46c5b807da3e715b93cb812c4584f2a06df86b951
-	sha512sums = 6950c072271251d1a2d1938a142004042e961cc3d36193c74c9b6eabcf00cb7e0cd7081095b972e8d5b515762d8493aef511e8d7d2691c23c3bdfc6034338c60
+	sha512sums = 2f5db53379fd4dadd60a81afe1d26cdea686decf1132ee781ab66ead2d19e85569511be7449e747eea5fc133c0865f56b996d86a66f3eb31f1697a24d6340030
 
 pkgname = petsc
diff --git a/PKGBUILD b/PKGBUILD
index 252ccf1..1ea8394 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -31,13 +31,17 @@ optdepends=('trilinos: support for trilinos'
   'triangle: support for Triangle'
   'trilinos-ml: support for ML (part of Trilinos)'
   'zoltan: support for zoltan'
+  'libyaml: support for YAML-formatted file'
+  'opencl-headers: support for OpenCL'
+  'ocl-icd: support for OpenCL'
+  'sundials: for CVODE solver support'
   )
 
 install=petsc.install
 source=(http://web.cels.anl.gov/projects/petsc/download/release-snapshots/${pkgname}-${pkgver}.tar.gz
         test_optdepends.sh)
 sha512sums=('68abd97377e1e30d3c09afa1617385e8ba56acc8508000c8da47112454462e9da35135076e678cea7a4fa6e46c5b807da3e715b93cb812c4584f2a06df86b951'
-            '6950c072271251d1a2d1938a142004042e961cc3d36193c74c9b6eabcf00cb7e0cd7081095b972e8d5b515762d8493aef511e8d7d2691c23c3bdfc6034338c60')
+            '2f5db53379fd4dadd60a81afe1d26cdea686decf1132ee781ab66ead2d19e85569511be7449e747eea5fc133c0865f56b996d86a66f3eb31f1697a24d6340030')
 
 _install_dir=/opt/petsc/${_config}
 _petsc_arch=arch-${_config}
diff --git a/test_optdepends.sh b/test_optdepends.sh
index 249b0de..a810e77 100644
--- a/test_optdepends.sh
+++ b/test_optdepends.sh
@@ -79,4 +79,15 @@ if [ -f "/usr/lib/libzoltan.so" ]; then
   CONFOPTS="${CONFOPTS} --with-zoltan=1"
 fi
 
+# Add OpenCL support
+OPENCL_DIR="/usr/include/CL"
+if [ -f "/usr/lib/libOpenCL.so" && -d "${OPENCL_DIR}" ]; then
+	CONFOPTS="${CONFOPTS} --with-opencl=1 --with-opencl-lib=-lOpenCL -with-opencl-include=${OPENCL_DIR}"
+fi
+
+# Add sundials support (complex-scalar is not supported)
+if [ -f "/usr/lib/libsundials_cvode.so" ]; then
+	CONFOPTS="${CONFOPTS} --with-sundials2=0"
+fi
+
 echo "${CONFOPTS}"
-- 
2.43.0

