From 8238f6bd7c0e3bf0d188dedd9b54260bee344b4f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Carlos=20Aznar=C3=A1n?= <caznaranl@uni.pe>
Date: Sun, 10 Mar 2024 11:06:31 -0500
Subject: [PATCH] Boost rebuild

---
 .SRCINFO | 2 +-
 PKGBUILD | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/.SRCINFO b/.SRCINFO
index 9719ded..0da1756 100644
--- a/.SRCINFO
+++ b/.SRCINFO
@@ -1,7 +1,7 @@
 pkgbase = vtk
 	pkgdesc = Software system for 3D computer graphics, image processing, and visualization
 	pkgver = 9.3.0
-	pkgrel = 10
+	pkgrel = 11
 	url = https://www.vtk.org
 	arch = x86_64
 	license = BSD-3-Clause
diff --git a/PKGBUILD b/PKGBUILD
index 9e8c317..ca358a8 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -8,7 +8,7 @@
 pkgname=vtk
 # May need bootstrapping on upgrades due to circular vtk <-> opencascade dependency
 pkgver=9.3.0
-pkgrel=10
+pkgrel=11
 pkgdesc="Software system for 3D computer graphics, image processing, and visualization"
 arch=(x86_64)
 url="https://www.vtk.org"
-- 
2.44.0

