From 5f1443a3f9ed2c653139654fc279492cbe44894a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Carlos=20Aznar=C3=A1n=20Laos?= <caznaranl@uni.pe>
Date: Mon, 10 Jul 2023 11:35:08 -0500
Subject: [PATCH] Bump trilinos version to 14.2.0

---
 .SRCINFO              |  8 ++++----
 PKGBUILD              |  9 +++++----
 compiler-errors.patch | 14 +-------------
 3 files changed, 10 insertions(+), 21 deletions(-)

diff --git a/.SRCINFO b/.SRCINFO
index 042efff..4acfd93 100644
--- a/.SRCINFO
+++ b/.SRCINFO
@@ -1,8 +1,8 @@
 pkgbase = trilinos
 	pkgdesc = algorithms for the solution of large-scale scientific problems
-	pkgver = 14.0.0
+	pkgver = 14.2.0
 	pkgrel = 1
-	url = http://trilinos.org
+	url = https://trilinos.org
 	arch = x86_64
 	license = LGPL3
 	checkdepends = cmake
@@ -21,9 +21,9 @@ pkgbase = trilinos
 	depends = hdf5-openmpi
 	provides = trilinos-sacado
 	provides = zoltan
-	source = https://github.com/trilinos/Trilinos/archive/refs/tags/trilinos-release-14-0-0.tar.gz
+	source = https://github.com/trilinos/Trilinos/archive/refs/tags/trilinos-release-14-2-0.tar.gz
 	source = compiler-errors.patch
-	sha256sums = 054d2fabdf70fce0dfaeb20eed265bd7894045d3e00c3d1ddb72d1c77c339ca1
+	sha256sums = c96606e5cd7fc9d25b9dc20719cd388658520d7cbbd2b4de77a118440d1e0ccb
 	sha256sums = 09546864a36ec83cfbedb29c97ead05778b4da370a84185565973f28f49f2880
 
 pkgname = trilinos
diff --git a/PKGBUILD b/PKGBUILD
index 4713a54..c146958 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -4,13 +4,14 @@
 # Contributor: Jingbei Li <i@jingbei.li>
 # Contributor: Simon Pintarelli <simon.pintarelli@gmail.com>
 # Contributor: Feng Wang <wanng.fenng@gmail.com>
+# Contributor: Carlos Aznarán <caznaranl@uni.pe>
 pkgname=trilinos
-pkgver=14.0.0
+pkgver=14.2.0
 _pkgver=${pkgver//./-}
 pkgrel=1
 pkgdesc="algorithms for the solution of large-scale scientific problems"
 arch=('x86_64')
-url="http://trilinos.org"
+url="https://trilinos.org"
 license=('LGPL3')
 depends=('python' 'lapack' 'boost' 'netcdf' 'libmatio' 'libx11' 'hdf5-openmpi')
 makedepends=('gcc-fortran' 'perl' 'blas' 'cmake' 'bc' 'python-numpy')
@@ -18,8 +19,8 @@ provides=('trilinos-sacado' 'zoltan')
 checkdepends=('cmake')
 source=("https://github.com/trilinos/Trilinos/archive/refs/tags/trilinos-release-$_pkgver.tar.gz"
         'compiler-errors.patch')
-sha256sums=('054d2fabdf70fce0dfaeb20eed265bd7894045d3e00c3d1ddb72d1c77c339ca1'
-            '09546864a36ec83cfbedb29c97ead05778b4da370a84185565973f28f49f2880')
+sha256sums=('c96606e5cd7fc9d25b9dc20719cd388658520d7cbbd2b4de77a118440d1e0ccb'
+            '1ad5df1f48db587f89f0c19584c1f37b7767857272f2dd01203fed48fff1bf81')
 
 prepare() {
   patch -d  Trilinos-trilinos-release-"$_pkgver" -p1 -i ../compiler-errors.patch
diff --git a/compiler-errors.patch b/compiler-errors.patch
index 3da154a..51b1db3 100644
--- a/compiler-errors.patch
+++ b/compiler-errors.patch
@@ -19,16 +19,4 @@ diff --color -urt Trilinos-trilinos-release-14-0-0.orig/packages/krino/krino/CMa
 -add_subdirectory(unit_tests)
  
  SET(SOURCES_MAIN Apps_krino.cpp)
- 
-diff --color -urt Trilinos-trilinos-release-14-0-0.orig/packages/teuchos/core/src/Teuchos_BigUIntDecl.hpp Trilinos-trilinos-release-14-0-0.new/packages/teuchos/core/src/Teuchos_BigUIntDecl.hpp
---- Trilinos-trilinos-release-14-0-0.orig/packages/teuchos/core/src/Teuchos_BigUIntDecl.hpp	2023-05-18 08:34:11.735903734 +0200
-+++ Trilinos-trilinos-release-14-0-0.new/packages/teuchos/core/src/Teuchos_BigUIntDecl.hpp	2023-05-18 08:29:49.596832958 +0200
-@@ -43,7 +43,7 @@
- #define TEUCHOS_BIG_UINT_DECL_HPP
- 
- #include <iosfwd>
--
-+#include <cstdint>
- /*! \file Teuchos_BigUIntDecl.hpp
-     \brief Arbitrary-precision unsigned integer declaration.
- */
+ 
\ No newline at end of file
-- 
2.41.0

