From 1a35f8c84656be8dfc6e5a63e065dc960429f840 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Carlos=20Aznar=C3=A1n=20Laos?= <caznaranl@uni.pe>
Date: Mon, 10 Jul 2023 12:19:51 -0500
Subject: [PATCH] Enable trilinos

---
 .SRCINFO | 2 +-
 PKGBUILD | 4 +++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/.SRCINFO b/.SRCINFO
index 1645b9a..514cacc 100644
--- a/.SRCINFO
+++ b/.SRCINFO
@@ -2,7 +2,7 @@ pkgbase = deal-ii
 	pkgdesc = An Open Source Finite Element Differential Equations Analysis Library
 	pkgver = 9.5.0
 	pkgrel = 1
-	url = http://www.dealii.org/
+	url = https://dealii.org
 	install = deal-ii.install
 	arch = i686
 	arch = x86_64
diff --git a/PKGBUILD b/PKGBUILD
index 4925a4c..fc2e193 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -7,7 +7,7 @@ pkgver=9.5.0
 pkgrel=1
 pkgdesc="An Open Source Finite Element Differential Equations Analysis Library"
 arch=("i686" "x86_64")
-url="http://www.dealii.org/"
+url="https://dealii.org"
 license=('LGPL')
 depends=('boost')
 optdepends=(
@@ -103,6 +103,8 @@ build() {
      cmake_configuration_flags+=" -DLAPACK_INCLUDE_DIRS=${MKLROOT}/include"
   fi
 
+  cmake_configuration_flags+=" -DDEAL_II_WITH_TRILINOS=ON"
+
   # For GSL compatibility we need the full link interface, which includes
   # libgslcblas, so disable --as-needed with GCC:
   sed -i '/ENABLE_IF_LINKS(DEAL_II_LINKER_FLAGS "-Wl,--as-needed")/d' \
-- 
2.41.0

