From fcccdf98552c43298ce5531046944f65ea273628 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Carlos=20Aznar=C3=A1n?= <caznaranl@uni.pe>
Date: Sat, 9 Nov 2024 19:10:35 -0500
Subject: [PATCH] Add support for OpenCL

---
 .SRCINFO           |  5 ++++-
 PKGBUILD           |  7 +++++--
 test_optdepends.sh | 11 +++++++++++
 3 files changed, 20 insertions(+), 3 deletions(-)

diff --git a/.SRCINFO b/.SRCINFO
index 49e1f1a..24d79c8 100644
--- a/.SRCINFO
+++ b/.SRCINFO
@@ -36,12 +36,15 @@ pkgbase = petsc-complex
 	optdepends = triangle: support for the two-dimensional quality mesh generator and Delaunay triangulator
 	optdepends = valgrind: support for valgrind to help find memory-management problems in programs
 	optdepends = zoltan: support for zoltan
+	optdepends = opencl-headers: support for OpenCL
+	optdepends = ocl-icd: support for OpenCL
+	optdepends = sundials: for CVODE solver support
 	provides = petsc=3.22.1
 	conflicts = petsc
 	options = staticlibs
 	source = https://web.cels.anl.gov/projects/petsc/download/release-snapshots/petsc-lite-3.22.1.tar.gz
 	source = test_optdepends.sh
 	sha512sums = 677038e89179307129b889f5c109bdec2d30a547fedf90fc3962ba35179f2d69bc40b4097a7ef867bbc3e334230523b8c4493b0c173184bc22255617bb542234
-	sha512sums = 6babb4f04fe8a503743cd0876cf75d8bcf2ebb1cfd2d89bcf10109721335b19f8b1b6ccb27cc8b85ea21597c2f1edd8f30824be614072a6498a6a92e39632468
+	sha512sums = d29a78f55e4c7ede37046601c75fd6fde6378064dabb6edab2d14ca2ebcd528be8a2308620e2109602ef453891a71e3fe1dcceb993cc43b93d363cf35792493e
 
 pkgname = petsc-complex
diff --git a/PKGBUILD b/PKGBUILD
index 7ee652b..25c539d 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -37,12 +37,15 @@ optdepends=('hypre: support for the hypre sparse system solver'
   'superlu_dist: support for the superlu_dist sparse solver'
   'triangle: support for the two-dimensional quality mesh generator and Delaunay triangulator'
   'valgrind: support for valgrind to help find memory-management problems in programs'
-  'zoltan: support for zoltan')
+  'zoltan: support for zoltan'
+  'opencl-headers: support for OpenCL'
+  'ocl-icd: support for OpenCL'
+  'sundials: for CVODE solver support')
 install=${_base}.install
 source=(https://web.cels.anl.gov/projects/${_base}/download/release-snapshots/${_base}-lite-${pkgver}.tar.gz
   test_optdepends.sh)
 sha512sums=('677038e89179307129b889f5c109bdec2d30a547fedf90fc3962ba35179f2d69bc40b4097a7ef867bbc3e334230523b8c4493b0c173184bc22255617bb542234'
-            '6babb4f04fe8a503743cd0876cf75d8bcf2ebb1cfd2d89bcf10109721335b19f8b1b6ccb27cc8b85ea21597c2f1edd8f30824be614072a6498a6a92e39632468')
+            'd29a78f55e4c7ede37046601c75fd6fde6378064dabb6edab2d14ca2ebcd528be8a2308620e2109602ef453891a71e3fe1dcceb993cc43b93d363cf35792493e')
 
 _install_dir=/opt/${_base}/${_config}
 _petsc_arch=arch-${_config}
diff --git a/test_optdepends.sh b/test_optdepends.sh
index 72aed27..77740de 100644
--- a/test_optdepends.sh
+++ b/test_optdepends.sh
@@ -90,4 +90,15 @@ if [ -f "/usr/lib/pkgconfig/valgrind.pc" ]; then
   CONFOPTS="${CONFOPTS} --with-valgrind=1"
 fi
 
+# Add OpenCL support
+OPENCL_DIR="/usr/include/CL"
+if [ -f "/usr/lib/libOpenCL.so" && -d "${OPENCL_DIR}" ]; then
+	CONFOPTS="${CONFOPTS} --with-opencl=1 --with-opencl-lib=-lOpenCL -with-opencl-include=${OPENCL_DIR}"
+fi
+
+# Add sundials support (complex-scalar is not supported)
+if [ -f "/usr/lib/libsundials_cvode.so" ]; then
+	CONFOPTS="${CONFOPTS} --with-sundials2=0"
+fi
+
 echo "${CONFOPTS}"
-- 
2.47.0

