From 98cc16c7e12aaec0c24059dcaa5081920567470c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Carlos=20Aznar=C3=A1n?= <caznaranl@uni.pe>
Date: Wed, 10 Jan 2024 14:36:52 -0500
Subject: [PATCH] Add support for OpenCL

---
 .SRCINFO           |  5 ++++-
 PKGBUILD           |  8 ++++++--
 test_optdepends.sh | 11 +++++++++++
 3 files changed, 21 insertions(+), 3 deletions(-)

diff --git a/.SRCINFO b/.SRCINFO
index 0b24a23..7f13060 100644
--- a/.SRCINFO
+++ b/.SRCINFO
@@ -31,12 +31,15 @@ pkgbase = petsc-complex
 	optdepends = suitesparse: support for the suitesparse sparse matrix libraries
 	optdepends = valgrind: support for valgrind to help find memory-management problems in programs
 	optdepends = libyaml: support for YAML-formatted file
+	optdepends = opencl-headers: support for OpenCL
+	optdepends = ocl-icd: support for OpenCL
+	optdepends = sundials: for CVODE solver support
 	provides = petsc=3.20.3
 	conflicts = petsc
 	options = staticlibs
 	source = https://web.cels.anl.gov/projects/petsc/download/release-snapshots/petsc-lite-3.20.3.tar.gz
 	source = test_optdepends.sh
 	sha512sums = 912538850eeaf6d78c090438f8198963cd787ef7ff3bbc841719b08fc738c7b20b7955850baacca4eada4a97b113492b9111d35afa33918ec52123e2f1a73f9b
-	sha512sums = 68bed9f836bf362b5d528af040afc2c572e87c43fac716d257862bcba06401a212adeec1fb84080a352831c22b968c9da65aa3eb5ef9e44be38ee0c0897c4b8d
+	sha512sums = f256044863abdf8f91d7d084c1df1a2d3487324f9f0b90709531b966d2881098d204af540fc3026647f0fb4116662ad2ffae042368d98a074eda60813767726b
 
 pkgname = petsc-complex
diff --git a/PKGBUILD b/PKGBUILD
index bb37ca9..9f8ff13 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -39,12 +39,16 @@ optdepends=('trilinos: support for zoltan'
   'triangle: support for the two-dimensional quality mesh generator and Delaunay triangulator'
   'suitesparse: support for the suitesparse sparse matrix libraries'
   'valgrind: support for valgrind to help find memory-management problems in programs'
-  'libyaml: support for YAML-formatted file')
+  'libyaml: support for YAML-formatted file'
+  'opencl-headers: support for OpenCL'
+  'ocl-icd: support for OpenCL'
+  'sundials: for CVODE solver support')
+
 install=${_base}.install
 source=(https://web.cels.anl.gov/projects/${_base}/download/release-snapshots/${_base}-lite-${pkgver}.tar.gz
   test_optdepends.sh)
 sha512sums=('912538850eeaf6d78c090438f8198963cd787ef7ff3bbc841719b08fc738c7b20b7955850baacca4eada4a97b113492b9111d35afa33918ec52123e2f1a73f9b'
-  '68bed9f836bf362b5d528af040afc2c572e87c43fac716d257862bcba06401a212adeec1fb84080a352831c22b968c9da65aa3eb5ef9e44be38ee0c0897c4b8d')
+  'f256044863abdf8f91d7d084c1df1a2d3487324f9f0b90709531b966d2881098d204af540fc3026647f0fb4116662ad2ffae042368d98a074eda60813767726b')
 
 _install_dir=/opt/${_base}/${_config}
 _petsc_arch=arch-${_config}
diff --git a/test_optdepends.sh b/test_optdepends.sh
index af84472..9c7a52c 100644
--- a/test_optdepends.sh
+++ b/test_optdepends.sh
@@ -120,4 +120,15 @@ if [ -f "/usr/lib/libyaml.so" ]; then
 	CONFOPTS="${CONFOPTS} --with-yaml=1"
 fi
 
+# Add OpenCL support
+OPENCL_DIR="/usr/include/CL"
+if [ -f "/usr/lib/libOpenCL.so" && -d "${OPENCL_DIR}" ]; then
+	CONFOPTS="${CONFOPTS} --with-opencl=1 --with-opencl-lib=-lOpenCL -with-opencl-include=${OPENCL_DIR}"
+fi
+
+# Add sundials support (complex-scalar is not supported)
+if [ -f "/usr/lib/libsundials_cvode.so" ]; then
+	CONFOPTS="${CONFOPTS} --with-sundials2=0"
+fi
+
 echo "${CONFOPTS}"
-- 
2.43.0

