From 678ad59dbfba9074b8be1fb76bbc38eceb0dcb86 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Carlos=20Aznar=C3=A1n?= <caznaranl@uni.pe>
Date: Thu, 29 Feb 2024 16:38:50 -0500
Subject: [PATCH] Add env

---
 .SRCINFO | 2 +-
 PKGBUILD | 6 +++++-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/.SRCINFO b/.SRCINFO
index 813eb5f..a818f8e 100644
--- a/.SRCINFO
+++ b/.SRCINFO
@@ -1,7 +1,7 @@
 pkgbase = dolfinx
 	pkgdesc = Next generation FEniCS problem solving environment
 	pkgver = 0.7.3
-	pkgrel = 1
+	pkgrel = 2
 	url = https://github.com/FEniCS/dolfinx
 	arch = x86_64
 	license = LGPL-3.0-or-later
diff --git a/PKGBUILD b/PKGBUILD
index 837e73c..008be0f 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -2,7 +2,7 @@
 pkgname=dolfinx
 pkgdesc="Next generation FEniCS problem solving environment"
 pkgver=0.7.3
-pkgrel=1
+pkgrel=2
 arch=(x86_64)
 url="https://github.com/FEniCS/${pkgname}"
 license=(LGPL-3.0-or-later GPL-3.0-or-later)
@@ -40,6 +40,10 @@ check() {
   ffcx ${pkgname}-${pkgver}/cpp/test/poisson.py -o ${pkgname}-${pkgver}/cpp/test
   DESTDIR="${PWD}/tmp_install" cmake --build build --target install
 
+  export PETSC_DIR=/opt/petsc/linux-c-opt
+  export SLEPC_DIR=/opt/slepc/linux-c-opt
+  env
+
   CMAKE_PREFIX_PATH="${srcdir}/tmp_install/usr/lib/cmake/${pkgname}" cmake \
     -S ${pkgname}-${pkgver}/cpp/test \
     -B build_test
-- 
2.44.0

