From 41384549943c0a3f22417f071a0dd1e8fdec9283 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Carlos=20Aznar=C3=A1n?= <caznaranl@uni.pe>
Date: Tue, 5 Mar 2024 23:51:53 -0500
Subject: [PATCH] Fix version

---
 .SRCINFO | 4 ++--
 PKGBUILD | 6 +++++-
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/.SRCINFO b/.SRCINFO
index 4ff4ead..e564039 100644
--- a/.SRCINFO
+++ b/.SRCINFO
@@ -4,8 +4,8 @@ pkgbase = fast_float
 	pkgrel = 1
 	url = https://github.com/fastfloat/fast_float
 	arch = any
-	license = Apache
-	license = Boost
+	license = Apache-2.0
+	license = BSL-1.0
 	license = MIT
 	makedepends = cmake
 	source = fast_float-6.1.0.tar.gz::https://github.com/fastfloat/fast_float/archive/refs/tags/v6.1.0.tar.gz
diff --git a/PKGBUILD b/PKGBUILD
index 7779f38..f874241 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -7,11 +7,15 @@ pkgrel=1
 pkgdesc='Fast and exact implementation of the C++ from_chars functions for float and double types'
 arch=('any')
 url="https://github.com/fastfloat/$pkgname"
-license=('Apache' 'Boost' 'MIT')
+license=('Apache-2.0' 'BSL-1.0' 'MIT')
 makedepends=('cmake')
 source=("$pkgname-$pkgver.tar.gz::$url/archive/refs/tags/v$pkgver.tar.gz")
 b2sums=('9c7abd293fcd0dd7e429825918b368071686d06baa97cee5d71099490dbed73acf5c8946f086dad21affaa54d94444a922d25d2ce9724a903944722790a7771e')
 
+prepare() {
+  sed -i "s|6.0.0|${pkgver}|" $pkgname-$pkgver/CMakeLists.txt
+}
+
 build() {
   cmake -B build -S $pkgname-$pkgver \
     -DCMAKE_BUILD_TYPE=Release \
-- 
2.44.0

