From 013dec1efb136a89bb2d1c0dcacf542abdc37f94 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Carlos=20Aznar=C3=A1n?= <caznaranl@uni.pe>
Date: Fri, 3 May 2024 19:52:29 -0500
Subject: [PATCH] Enable python bindings

---
 .SRCINFO |  5 ++++-
 PKGBUILD | 10 +++++++---
 2 files changed, 11 insertions(+), 4 deletions(-)

diff --git a/.SRCINFO b/.SRCINFO
index 3e059d2..dca34d9 100644
--- a/.SRCINFO
+++ b/.SRCINFO
@@ -1,11 +1,14 @@
 pkgbase = libisl
 	pkgdesc = Library for manipulating sets and relations of integer points bounded by linear constraints
 	pkgver = 0.26
-	pkgrel = 2
+	pkgrel = 3
 	url = https://libisl.sourceforge.io
 	arch = x86_64
 	license = MIT
+	makedepends = llvm
+	makedepends = clang
 	depends = gmp
+	depends = python
 	provides = isl
 	provides = libisl.so
 	replaces = isl
diff --git a/PKGBUILD b/PKGBUILD
index 6391106..2690a76 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -6,12 +6,13 @@
 
 pkgname=libisl
 pkgver=0.26
-pkgrel=2
+pkgrel=3
 pkgdesc='Library for manipulating sets and relations of integer points bounded by linear constraints'
 arch=('x86_64')
 url='https://libisl.sourceforge.io'
 license=('MIT')
-depends=('gmp')
+depends=('gmp' 'python')
+makedepends=('llvm' 'clang')
 provides=('isl' 'libisl.so')
 replaces=('isl')
 options=(staticlibs)
@@ -20,7 +21,7 @@ b2sums=('2cfe2e0774a991d2385e66f58490c5454382cabc82d74eb4bbe5a603293b7bdc34ca3b0
 
 build() {
   cd "${srcdir}"/${pkgname#lib}-${pkgver}
-  ./configure --prefix=/usr
+  ./configure --prefix=/usr --with-clang=system
   make
 }
 
@@ -34,6 +35,9 @@ package() {
 
   make DESTDIR="${pkgdir}" install
 
+  local _sitepackages=$(python -c "import site; print(site.getsitepackages()[0])")
+  install -Dm644 interface/${pkgname#lib}.py -t "${pkgdir}${_sitepackages}"
+
   install -dm755 "${pkgdir}"/usr/share/gdb/auto-load/usr/lib/
   mv "${pkgdir}"/usr/lib/libisl.so.*-gdb.py "${pkgdir}"/usr/share/gdb/auto-load/usr/lib/
 
-- 
2.45.0

